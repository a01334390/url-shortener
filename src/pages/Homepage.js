import React from 'react'
import { Window, WindowContent, WindowHeader, Button, Anchor, TextArea, Hourglass } from "react95"

// Style JSON sheet
const style = {
    window: {
        position:'fixed',
        top:'50%',
        left:'50%',
        transform: 'translate(-50%,-50%)',
        width: '400px'
    },
    windowHeader : {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
    },
    buttonmenu: {
        marginRight: '-6px',
        marginTop: '1px' 
    },
    button: {
        marginTop: '3px',
        left: '130px'
    },
    span: {
        fontWeight: 'bold', 
        transform: 'translateY(-1px)'
    },
    h1: {
        fontFamily: 'times new roman',
        textAlign: 'center',
        fontSize: '3rem',
        marginTop: '0.5rem',
    },
    p: {
        fontFamily: 'times new roman',
        textAlign: 'center',
        marginBottom: '5px'
    },
    textarea: {
        marginTop: '3px'
    }
}

class App extends React.Component {
    state = {
        submitState : false,
        link : ''
    }

    onSubmit = () => {
        this.setState({submitState: true})
    }

    onWindowClose = () => {
        window.open("about:blank", "_self");
        window.close();
    }
    
    render() {
        return (
            <Window style={style.window }>
                <WindowHeader style={style.windowHeader}>
                    <span>URL Shortener</span>
                    <Button onClick={this.onWindowClose} style={style.buttonmenu} size={'sm'} square>
                        <span style={style.span}>x</span>
                    </Button>
                </WindowHeader>
                <WindowContent>
                    <h1 style={style.h1}>
                         URL Shortener 
                    </h1>
                    <p style={style.p}> Created by <Anchor href="https://www.github.com/a01334390" target="_blank"> A01334390</Anchor></p>
                    <TextArea disabled={this.state.submitState} onChange={(event) => this.setState({link: event.target.value})} style={style.textarea} shadow={true} placeholder="Place your link here..."></TextArea>
                    <div style={style.button}>
                    {!this.state.submitState ? 
                        <Button onClick={this.onSubmit} style={style.button}>
                             Convert 
                        </Button> :
                        <Hourglass/>
                    }
                    </div>
                </WindowContent>
            </Window>
        )
    }
}

export default App