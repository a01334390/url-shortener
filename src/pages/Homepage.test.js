import React from 'react';
import { shallow, mount } from 'enzyme';

import Homepage from './Homepage'

describe('Homepage testing',() => {
    it('should render correctly in "Debug" mode',()=>{
        const component = shallow(<Homepage debug/>)
        expect(component).toMatchSnapshot()
    })

    it('should disable textarea when submit button is clicked',()=>{
        const component = mount(<Homepage/>)
        component
            .find('Button')
            .at(1)
            .simulate('click')
        const textarea = component.find('TextArea')
        expect(textarea.props().disabled).toEqual(true)
    })

    it('should make the Hour Glass Component visible', () => {
        const component = mount(<Homepage/>)
        component
            .find('Button')
            .at(1)
            .simulate('click')
        const hourglass = component.find('Hourglass')
        expect(hourglass.props().disabled).toEqual(undefined)
    })

    it('should create a blank page when x button is pressed',() => {
        const component = mount(<Homepage/>)
        component
            .find('Button')
            .at(0)
            .simulate('click')
        let window = component
            .find('Window')
        expect(window).toEqual({})
    })
})