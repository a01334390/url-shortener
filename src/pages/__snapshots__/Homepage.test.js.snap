// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage testing should render correctly in "Debug" mode 1`] = `
<Fragment>
  <Window
    shadow={true}
    style={
      Object {
        "left": "50%",
        "position": "fixed",
        "top": "50%",
        "transform": "translate(-50%,-50%)",
        "width": "400px",
      }
    }
  >
    <WindowHeader
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <span>
        URL Shortener
      </span>
      <Button
        active={false}
        disabled={false}
        fullWidth={false}
        onClick={[Function]}
        size="sm"
        square={true}
        style={
          Object {
            "marginRight": "-6px",
            "marginTop": "1px",
          }
        }
        type="button"
        variant="default"
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
              "transform": "translateY(-1px)",
            }
          }
        >
          x
        </span>
      </Button>
    </WindowHeader>
    <WindowContent>
      <h1
        style={
          Object {
            "fontFamily": "times new roman",
            "fontSize": "3rem",
            "marginTop": "0.5rem",
            "textAlign": "center",
          }
        }
      >
        URL Shortener
      </h1>
      <p
        style={
          Object {
            "fontFamily": "times new roman",
            "marginBottom": "5px",
            "textAlign": "center",
          }
        }
      >
         Created by 
        <Anchor
          href="https://www.github.com/a01334390"
          target="_blank"
        >
           A01334390
        </Anchor>
      </p>
      <TextArea
        disabled={false}
        onChange={[Function]}
        placeholder="Place your link here..."
        shadow={true}
        style={
          Object {
            "marginTop": "3px",
          }
        }
        variant="default"
      />
      <div
        style={
          Object {
            "left": "130px",
            "marginTop": "3px",
          }
        }
      >
        <Button
          active={false}
          disabled={false}
          fullWidth={false}
          onClick={[Function]}
          size="md"
          square={false}
          style={
            Object {
              "left": "130px",
              "marginTop": "3px",
            }
          }
          type="button"
          variant="default"
        >
          Convert
        </Button>
      </div>
    </WindowContent>
  </Window>
</Fragment>
`;
